### DISCUSSION TITLE
"Cannot re-use the linked column name for a linked table."

### DISCUSSION THREAD

**Discussion by gemmell** [Budibase/budibase#3512] [Help] [answered]
I am evaluating Budibase and have started by just using the inbuilt database. My data structure requires a table called "Roles" which has a column name "Name" and a table "Meetings" where one column is "Required Participants" and is supposed to be a list of the Relationships with the Role table, and a second is "Optional Participants" which is also a list of Relationships with the Role table.

When adding the second relationship budibase doesn't add it, and doesn't give any errors. Only when I open the network tab in the chrome debugging tools and have a look at the call do I see the error message on the 400 response:
message: "Cannot re-use the linked column name for a linked table."

![image](https://user-images.githubusercontent.com/3324817/143146058-47a7dbe0-f4a8-4cc7-804d-225edb42b461.png)

There's two issues here:
A) As you can see from the screenshot, it just drops me back at the "Create your first table to start building" screen with no visible error message. 
B) Why can't I create two columns that are relationships to the same table? This is a pretty common use case for regular relational databases. I might have a Person table with teachers and students in it, and a Class table with the rows Teacher, Students... maybe not the greatest decomposition but you get the point. Two links to the same table from the same table...

**Comment by mjashanks** [comment, marked as answer]
Hi @gemmell

(I just deleted my last response, so I don't confuse others)

Thanks for reporting this - we should report a proper error message here, this is not a good experience.

It's currently possible to create 2 of the same relationship on the same table. 

When you create a relationship, it creates a column on **both** tables, so you can see the reverse relationship.

What has probably happened here is that the `Column name in other table` has defaulted to the same name as an existing column (see red box below). Just give it a new name.

![image](https://user-images.githubusercontent.com/3524181/143253021-1ff95df0-52b7-46cf-ad29-fbc1651b82bd.png)

**Comment by gemmell** [comment]
I see, so "Column name in other table" is pretty confusing then. Because I come from a dev background, I expect to be able to put in the column to join on. i.e. I have an ID column in Role and then have two Relationships in this table to the Role table using the ID. But actually that's not what it is at all. When I put the Column name in the other table it's just a backwards relationship rather than a join. You might want to make that clearer because it'll be confusing to anyone coming from a dev background.
